rfcs/0000-upgradeable-lifecycle.md

```diff
@@ -0,0 +1,75 @@
+**Problem:**
+Currently, the lifecycle version used in kpack is not configurable, nor is it user upgradeable. Users must wait for a new kpack release to be able to upgrade the version of the lifecycle used by their builders.
+
+**Outcome:**
+We would like to decouple the lifecycle upgrade process from the kpack upgrade process, allowing users to upgrade the lifecycle on their own.
+
+**Actions to take:**
+We would create a ClusterLifecycle resource similiar a store or stack. The resources would be tied to each builder so that different builders can have different lifecycle versions. This has the benefit of allowing users to test new lifecycle versions without having to spin up another cluster. Also, users who created their own buildpack would not have to worry about a newer lifecycle dropping support for the Buildpack Api they are using. Due to the operator focused nature of this resource, and its similarity to ClusterStores and ClusterStacks, implementing this as a cluster scoped resource makes more sense at this time.
```

Comment:
> Also, users who created their own buildpack would not have to worry about a newer lifecycle dropping support for the Buildpack Api they are using

I think realistically this should not be concern as the lifecycle should maintain backwards compatibility to old buildpack api versions.